search/searcher/search_intersects_test.go

```diff
@@ -92,6 +92,154 @@ func TestPointIntersects(t *testing.T) {
 	}
 }
 
+func TestPointLinestringIntersects(t *testing.T) {
+	tests := []struct {
+		QueryShape       []float64
+		DocShapeVertices [][]float64
+		DocShapeName     string
+		Desc             string
+		Expected         []string
+	}{
+		{
+			QueryShape:       []float64{4.0, 4.0},
+			DocShapeVertices: [][]float64{{2.0, 2.0}, {3.0, 3.0}, {4.0, 4.0}},
+			DocShapeName:     "linestring1",
+			Desc:             "point at the vertex of linestring",
+			Expected:         []string{"linestring1"},
+		},
+		{ // check this one
+			QueryShape:       []float64{1.5, 1.5001714},
+			DocShapeVertices: [][]float64{{0.0, 0.0}, {1.0, 1.0}, {2.0, 2.0}, {3.0, 3.0}},
+			DocShapeName:     "linestring1",
+			Desc:             "point along linestring",
+			Expected:         []string{"linestring1"},
+		},
+		{ // check this one
+			QueryShape:       []float64{1.5, 1.6001714},
+			DocShapeVertices: [][]float64{{0.0, 0.0}, {1.0, 1.0}, {2.0, 2.0}, {3.0, 3.0}},
+			DocShapeName:     "linestring1",
+			Desc:             "point outside linestring",
+			Expected:         nil,
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		doc := document.NewDocument(test.DocShapeName)
+		doc.AddField(document.NewGeoShapeFieldWithIndexingOptions("geometry", []uint64{},
+			[][][][]float64{{test.DocShapeVertices}}, "multipoint", document.DefaultGeoShapeIndexingOptions))
+		err := i.Update(doc)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		indexReader, err := i.Reader()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapePointRelationQuery("intersects",
+				false, indexReader, [][]float64{test.QueryShape}, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for point: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+		err = i.Delete(doc.ID())
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+		err = indexReader.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestPointPolygonIntersects(t *testing.T) {
+	tests := []struct {
+		QueryShape       []float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Desc             string
+		Expected         []string
+	}{
+		{ // check this one
```

Comment:
@sreekanth-cb these three cases were the ones not matching. Please let me know if this divergence is expected.
Thanks!